### DISCUSSION TITLE
5V external relay board

### DISCUSSION THREAD

**Discussion by JJTechman68** [phil-barrett/PicoCNC#10] [Q&A] [answered]
Hi Phil. I am wiring up the board now, and wanted to use an external 5v relay board. It requires signal pins and a constant 5V and ground. Just wondering where I would pull the constant 5V from and whether I should use the 5v/SSR section or the relay section of the PicoCNC board for my signal pins.

Also I am still struggling with how I set up a basic MPG pendant. I bought a daughter card for the I2C interface but am stuck at this point. Any help from anyone on this would be very much appreciated.

**Comment by phil-barrett** [comment]
It depends on the relay board.  could you send me a photo or link to it?

I am not sure about your I2C board, do you have a link for more info on it?

**Comment by JJTechman68** [comment]
Thanks for the quick reply!
Attached are images of the 2 boards. The relay board is just a simple
Amazon special.

Thanks

Jim

**Reply by phil-barrett** [reply]
I don't see any attachments. I'm guessing it is a powered relay board.  I don't recommend powering a relay coil from the Pico.  Firing the relay will cause voltage disruption that can lead to logic glitches or even a Pico reboot. 

You would use the TTL/SSR pins to control your external relay board.  Make sure the relay board and the PicoCNC share a common ground.

Have you see my [blog on using relays](https://www.grbl.org/single-post/using-relays-with-grblhal-and-the-t41u5xbb)?

**Comment by JJTechman68** [comment]
Sorry Phil, I answered by email and not through so they must have got lost. I've attached them here.
Since I'm using 2 separate 12V supplies to power the board as you recommend, should I ensure the ground is common with the 12V supply that's on the 5V TTL/SSR side of the board?
Also, I am using 3 wire, PNP 6-36V proximity sensors, so I'm assuming I need to have a common 12v positive and then use the sig pin for the switched wire?
![3-wire_proximity_sensor](https://github.com/phil-barrett/PicoCNC/assets/76827669/5dd67bd0-9eb6-4386-b7ec-189478f88624)
![4_channel-relay](https://github.com/phil-barrett/PicoCNC/assets/76827669/869e6a89-0942-4887-909f-16ba03e03139)
![I2C_expansion-board](https://github.com/phil-barrett/PicoCNC/assets/76827669/61b440e1-5976-4144-9b2d-e860d0f05384)
![I2C_expansion-board_specs](https://github.com/phil-barrett/PicoCNC/assets/76827669/1f84ea14-a2c7-49e7-a8da-b6dedc1e3c71)

**Comment by JJTechman68** [comment]
How do you select ganged and autosquared for the y axis motors in the WebBuilder? I am using the PICOCNC which is sooooo close to operating fully.

**Reply by phil-barrett** [reply]
The pull-down on the axis selection does it.
![image](https://github.com/phil-barrett/PicoCNC/assets/16570958/cb02e304-5e40-40a6-99ee-58e3c739036f)

**Reply by JJTechman68** [reply]
Does it select ganged and autosquared together for that axis? Your blog mentioned you need both options selected.

**Reply by phil-barrett** [reply, marked as answer]
There are 3 options for each axis - Single motor, Ganged motors and Auto-squared motors.  The Auto-square option includes Ganged motors.

**Reply by JJTechman68** [reply]
Thanks!
